osquery/tables/system/linux/kernel_keys.cpp

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+#include <fstream>
+
+#include <boost/algorithm/string/trim.hpp>
+
+#include <osquery/core/tables.h>
+#include <osquery/filesystem/filesystem.h>
+#include <osquery/logger/logger.h>
+#include <osquery/utils/conversions/split.h>
+
+namespace osquery {
+namespace tables {
+
+static const std::string kProcKeysPath = "/proc/keys";
+
+QueryData genKernelKeys(QueryContext& context) {
+  if (!pathExists(kProcKeysPath).ok()) {
+    VLOG(1) << "Cannot find proc keys file: " << kProcKeysPath;
+    return {};
+  }
+
+  std::ifstream fd(kProcKeysPath, std::ios::in);
```

Comment:
This looks pretty clean, but please use the `read_file` helper.

Reply:
https://github.com/osquery/osquery/blob/f15b05a48c43f2fa67609b5f47a7468a1bffdf1e/osquery/filesystem/filesystem.cpp#L226

Reply:
Fixed, Thanks for the comment